post_office/sanitizer.py

```diff
@@ -0,0 +1,145 @@
+            'tbody': ['class', 'id', 'style'],
+            'tfoot': ['class', 'id', 'style'],
+            'td': ['class', 'id', 'style', 'abbr', 'align', 'bgcolor', 'colspan', 'dir', 'height', 'lang', 'rowspan', 'scope', 'style', 'valign', 'width'],
+            'th': ['class', 'id', 'style', 'abbr', 'align', 'background', 'bgcolor', 'colspan', 'dir', 'height', 'lang', 'scope', 'style', 'valign', 'width'],
+            'thead': ['class', 'id', 'style'],
+            'tr': ['class', 'id', 'style', 'align', 'bgcolor', 'dir', 'style', 'valign'],
+            'u': ['class', 'id', 'style'],
+            'ul': ['class', 'id', 'style', 'dir', 'type'],
+        },
+        strip=True,
+        strip_comments=True,
+        styles=[
+            'border', 'border-top', 'border-right', 'border-bottom', 'border-left',
+            'border-radius',
+            'box-shadow',
+            'height',
+            'margin', 'margin-top', 'margin-right', 'margin-bottom', 'margin-left',
+            'padding', 'padding-top', 'padding-right', 'padding-bottom', 'padding-left',
+            'width',
+            'max-width',
+            'min-width',
+            'border-collapse',
+            'border-spacing',
+            'caption-side',
+            'empty-cells',
+            'table-layout',
+            'direction',
+            'font',
+            'font-family',
+            'font-style',
+            'font-variant',
+            'font-size',
+            'font-weight',
+            'letter-spacing',
+            'line-height',
+            'text-align',
+            'text-decoration',
+            'text-indent',
+            'text-overflow',
+            'text-shadow',
+            'text-transform',
+            'white-space',
+            'word-spacing',
+            'word-wrap',
+            'vertical-align',
+            'color',
+            'background',
+            'background-color',
+            'background-image',
+            'background-position',
+            'background-repeat',
+            'bottom',
+            'clear',
+            'cursor',
+            'display',
+            'float',
+            'left',
+            'opacity',
+            'outline',
+            'overflow',
+            'position',
+            'resize',
+            'right',
+            'top',
+            'visibility',
+            'z-index',
+            'list-style-position',
+            'list-style-tyle',
+        ],
+    ))
+except ImportError:
+    # if bleach is not installed, handle rendered HTML as plain text
+    heading = gettext_lazy("Stripping all HTML tags – install 'bleach' to render HTML properly.")
+    clean_html = lambda body: mark_safe(format_lazy('<p><em>({heading})</em></p><div>{body}</div>',
+                                                    heading=heading,
+                                                    body=escape(strip_tags(body))))
```

Comment:
If we're already calling `escape`, HTML tags will be displayed as regular text (e.g `<p>A paragraph</p>`).

Since the purpose of this feature is to give a preview of the rendered content, it'd be better if we lose the `strip_tags` so the HTML source code is properly output.

After `strip_tags` is removed, we can merge this in.

Further improvement to this would be to actually divide the HTML preview into two sections:
1. The actual source code preview
2. The rendered HTML, which only shows if `bleach` is installed.

But I think the improvement can come at a later pull request. This PR has been a long time coming and I also want to see this merged in :).

Reply:
Now if `bleach` is installed, emails are rendered as well readable HTML. If bleach is missing, HTML is rendered, but escaped making it hard to read.

> Further improvement to this would be to actually divide the HTML preview into two sections:

I'm not so convinced about this. Users of the Django admin backend do not necessarily have a technical background and "seeing" escaped HTML often does not make sense for them.